### PR TITLE
React to Auth/Options changes

### PR DESCRIPTION
**Pull request by HaoK**
Not a huge change to startup

Thoughts? @divega

### CODE REVIEW

**Comment by HaoK** [comment]
One bit of smell is the static defaults I moved to IdentityCookieOptions, those strings are what we default the AuthenticationSchemes to for the cookie middleware we configure.  Those are then copied to the options, so we have mutable strings.  Its just a bit wonky.

File: src/Microsoft.AspNet.Identity/IdentityOptions.cs [hunk 1]
```diff
@@ -83,52 +83,27 @@ public class IdentityOptions
...
-        /// <summary>
-        /// Gets or sets the authentication type used when constructing an <see cref="ClaimsIdentity"/> from an application cookie.
-        /// </summary>
-        /// <value>The authentication type used when constructing an <see cref="ClaimsIdentity"/> from an application cookie.</value>
-        public static string ApplicationCookieAuthenticationType { get; set; } = typeof(IdentityOptions).Namespace + ".Application.AuthType";
+        public string TwoFactorUserIdCookieAuthenticationScheme { get; set; } = IdentityCookieOptions.TwoFactorUserIdCookieAuthenticationScheme;
 
         /// <summary>
-        /// Gets or sets the authentication type used when constructing an <see cref="ClaimsIdentity"/> from an external identity cookie.
+        /// Gets or sets the scheme used to identify Two Factor authentication cookies for saving the Remember Me state,, must match the scheme in IdentityCookieOptions
```
Comment (divega) [inline src/Microsoft.AspNet.Identity/IdentityOptions.cs]: double comma?

**Comment by divega** [comment]
Something doesn't feel quite right but it is hard for me to explain it. It might be that I just need to understand how it works :smile: Can we go over it on a whiteboard?

**Comment by HaoK** [comment]
Closing and will resubmit with new iteration

**Pull request closed without merging**